upcloud/resource_upcloud_objectstorage.go

```diff
@@ -0,0 +1,205 @@
+)
+
+func resourceUpCloudObjectStorage() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceObjectStorageCreate,
+		ReadContext:   resourceObjectStorageRead,
+		UpdateContext: resourceObjectStorageUpdate,
+		DeleteContext: resourceObjectStorageDelete,
+		Schema: map[string]*schema.Schema{
+			"size": {
+				Description:  "The size of the object storage bucket in gigabytes",
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntInSlice([]int{250, 500, 1000}),
+			},
+			"access_key": {
+				Description:  "The access key used to identify user",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(4, 255),
+			},
+			"secret_key": {
+				Description:  "The secret key used to authenticate user",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(8, 255),
+			},
+			"zone": {
+				Description: "The zone in which the object storage bucket will be created",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"name": {
+				Description: "The name of the object storage bucket to be created",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			"description": {
+				Description: "The description of the object storage bucket to be created",
+				Optional:    true,
+				Type:        schema.TypeString,
+				ValidateFunc: validation.StringLenBetween(1, 255),
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"used_space": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceObjectStorageCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var (
+	    diags diag.Diagnostics
+		req request.CreateObjectStorageRequest
+	)
+
+	client := m.(*service.Service)
+
+	req.Size = d.Get("size").(int)
+	req.Zone = d.Get("zone").(string)
+	req.AccessKey = d.Get("access_key").(string)
+	req.SecretKey = d.Get("secret_key").(string)
+
+	if v, ok := d.GetOk("name"); ok {
+		req.Name = v.(string)
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		req.Description = v.(string)
+	}
+
+	objStorage, err := client.CreateObjectStorage(&req)
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to create object storage",
+			Detail:   err.Error(),
+		})
+		return diags
+	}
+
+	d.SetId(objStorage.UUID)
+
+	copyObjectStorageDetails(objStorage, d)
+
+	return diags
+}
+
+func resourceObjectStorageRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*service.Service)
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
```

Comment:
we're not doing this thou. Not sure if it makes any sense to have it set up like this, if we want to just throw the error without continuing further

Any thoughts?